src/nebula/NebulaAction.cpp

```diff
@@ -344,6 +358,173 @@ ResultCode CheckLeadersAction::checkResp(const ExecutionResponse& resp) const {
+        if (leaderStr.empty()) {
+            return ResultCode::ERR_FAILED;
+        }
+
+        std::vector<folly::StringPiece> spaceLeaders;
+        folly::split(",", leaderStr, spaceLeaders);
+        for (auto& sl : spaceLeaders) {
+            std::vector<folly::StringPiece> oneSpaceLeaderPair;
+            folly::split(":", sl, oneSpaceLeaderPair);
+            CHECK_EQ(2, oneSpaceLeaderPair.size());
+            auto spaceName = folly::trimWhitespace(oneSpaceLeaderPair[0]);
+            if (spaceName.str() == spaceName_) {
+                auto leaderNum = folly::trimWhitespace(oneSpaceLeaderPair[1]);
+                if (!first) {
+                    restult_ += ",";
+                }
+                restult_ += ipStr;
+                restult_ += ":";
+                restult_ += leaderNum.str();
+                first = false;
+                break;
+            }
+        }
+    }
+    return ResultCode::OK;
+}
+
+ResultCode CheckLeadersAction::doRun() {
+    CHECK_NOTNULL(client_);
+    auto cmd = command();
+    LOG(INFO) << "Send " << cmd << " to graphd";
+    ExecutionResponse resp;
+    int32_t retry = 0;
+    while (++retry < retryTimes_) {
+        client_->execute(cmd, resp);
+        LOG(INFO) << "Execute " << cmd << " successfully!";
+        auto ret = checkResp(resp);
+        if (ret == ResultCode::ERR_FAILED_NO_RETRY) {
+            return ret;
+        }
+        if (ret == ResultCode::OK) {
+            if (resultVarName_.empty()) {
+                return ResultCode::OK;
+            } else {
+                ret = checkLeaderDis(resp);
+                if (ret == ResultCode::ERR_FAILED_NO_RETRY) {
+                    return ret;
+                }
+                if (ret == ResultCode::OK) {
+                    ctx_->exprCtx.setVar(resultVarName_, std::move(restult_));
+                    return ret;
+                }
+            }
+        }
+        LOG(ERROR) << "Execute " << cmd << " failed, retry " << retry
+                   << " after " << retry << " seconds...";
+        sleep(retry);
+    }
+    return ResultCode::ERR_FAILED;
+}
+
+ResultCode SetFlagAction::buildCmd() {
+    folly::toLowerAscii(layer_);
+    folly::toLowerAscii(name_);
+    folly::toLowerAscii(value_);
+    if (layer_ != "storage" && layer_ != "meta" && layer_ != "graph") {
+       return ResultCode::ERR_FAILED;
+    }
+
+    cmd_ ="update configs ";
+    cmd_ += layer_;
+    cmd_ += ":";
+    if (name_ == "disable_auto_compactions")  {
+        cmd_ += "rocksdb_column_family_options = { disable_auto_compactions = ";
+        if (value_ != "true" && value_ != "false") {
+            LOG(ERROR) << "Failed to get rocksdb_column_family_options value";
+            return ResultCode::ERR_FAILED;
+        }
+        cmd_ += value_;
+        cmd_ += " } ";
+    } else if (name_ == "wal_ttl") {
+        try {
+            auto ttlVal = folly::to<int32_t>(value_);
+            value_ = folly::to<std::string>(ttlVal);
```

Comment:
Why convert it back?

Reply:
No need to onvert it back.
Here only check the validity of value

Reply:
has fixed